prefect_docker/containers.py

```diff
@@ -146,3 +146,45 @@ def start_docker_container_flow():
         await run_sync_in_worker_thread(container.start, **start_kwargs)
 
     return container
+
+
+@task
+async def stop_docker_container(
+    container_id: str,
+    docker_host: Optional[DockerHost] = None,
+    **stop_kwargs: Dict[str, Any],
+) -> Container:
+    """
+    Stops a container. Similar to the docker stop command.
+
+    Args:
+        container_id: The container ID to stop.
+        docker_host: Settings for interacting with a Docker host.
+        **stop_kwargs: Additional keyword arguments to pass to
+            [`client.containers.get(container_id).stop`](https://docker-py.readthedocs.io/en/stable/containers.html#docker.models.containers.Container.stop).
+
+    Returns:
+        The Docker Container object.
+
+    Examples:
+        Stop a container with an ID that starts wtih "c157".
+        ```
+        from prefect import flow
+        from prefect_docker.containers import stop_docker_container
+
+        @flow
+        def stop_docker_container_flow():
+            container = stop_docker_container(container_id="c157")
+            return container
+
+        stop_docker_container_flow()
+        ```
+    """
+    logger = get_run_logger()
+
+    with (docker_host or DockerHost()).get_client() as client:
+        logger.info(f"Stopping {container_id!r} container.")
+        container = await run_sync_in_worker_thread(client.containers.get, container_id)
```

Comment:
I think it'd be a good idea to log the full container ID once we have it for increased excitivity.
```suggestion
        container = await run_sync_in_worker_thread(client.containers.get, container_id)
        logger.info(f"Stopping container {container.id!r}")
```